vmm_core/virt_support_x86emu/src/emulate.rs

```diff
@@ -777,10 +803,81 @@ impl<T: EmulatorSupport, U: CpuIo> x86emu::Cpu for EmulatorCpu<'_, T, U> {
         Ok(())
     }
 
-    fn get_xmm(&mut self, reg: usize) -> Result<u128, Self::Error> {
-        self.support.get_xmm(reg).map_err(Error::Hypervisor)
+    fn gp(&mut self, reg: RegisterIndex) -> u64 {
+        let extended_register = self.support.gp(reg.extended_index);
+
+        (match reg.size {
```

Comment:
If reg.size becomes an enum then you can remove the panic branch, and store the shift field only when the size is 1.